### DISCUSSION TITLE
Unable to OAuth using React-Native (Expo)

### DISCUSSION THREAD

**Discussion by kingdevnl** [pocketbase/pocketbase#2958] [Q&A]
https://github.com/kingdevnl/expo-pocketbase-auth/blob/main/App.tsx

`
 LOG  Logging in
 LOG  Error
 LOG  [ClientResponseError 0: Something went wrong while processing your request.]
 LOG  [Error: EventSource connect took too long.]
`
It seems that it's unable to create the SSE connection.

**Comment by ganigeorgiev** [comment]
On what environment you are testing?

Is the above error reproducible in https://snack.expo.dev/?

**Reply by kingdevnl** [reply]
Windows is my PC, but the app is running on Android 13 (Expo 49)
I haven't used expo snack before so I'm not sure on how to import a git repo into it

**Reply by ganigeorgiev** [reply]
You can import a git repo from the Expo sidebar panel:
![sidebar](https://i.imgur.com/a7kGnWy.png)

Are you sure that the PocketBase API server is accessible from the Android device?

Do regular collection requests resolve properly?

Is there any specific reason why this import is like `import registerRootComponent from 'expo/build/launch/registerRootComponent';` and not like:

```js
import { registerRootComponent } from 'expo'
```
?

_I'll try to test the repo a little bit later with the Android emulator._

**Comment by ganigeorgiev** [comment]
@kingdevnl I'm not able to reproduce the issue in the snack.expo emulator and via Expo Go on my Android (LineageOS) 7 (_I don't have access to more modern device at the moment_).

Your repo has several dependency issues, so I've created a new minimal one in https://snack.expo.dev/F7CKHi9WT.

The above is using a `gitlab` as provider but it should be the same with `github`.

I'm also using [ngrok](https://ngrok.com/) (eg. `ngrok http 8090`) to expose my PocketBase localhost server so that I can access it from the device outside of my network.

If you are still not able to resolve your issue, then you could:
- Make sure that your device has access to the PocketBase server (if not, use ngrok or other similar tool to create a public address).
- Review again the comments in https://github.com/pocketbase/pocketbase/discussions/2484 and try to disable flipper or any other debug mode utility

If none of the above helps, then you can try to experiment with some other SSE polyfill.

_In general there are plans to remove the `EventSource` dependency with the JS SDK refactoring and replace it with manual parsed `fetch` stream, but that will be after v0.17.0._

**Reply by kingdevnl** [reply]
I'm quite sure the issue for me is that flipper is breaking it, I'm not sure how to disable it in expo where the project is managed and not ejected

**Comment by mwaleed-39** [comment]
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, View } from 'react-native';
import React from 'react';
import Reg_form from './Frontend/Components/RegForm';
import Login_form from './Frontend/Components/Signin';
import ForgotPassword from './Frontend/Components/ForgotPassword';
import { ApplicationProvider, Layout, Text } from '@ui-kitten/components';
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import {
createDrawerNavigator,
DrawerContentScrollView,
DrawerItemList,
DrawerItem,
} from '@react-navigation/drawer';
import * as eva from '@eva-design/eva';
import ChangePassword from './Frontend/Components/ChangePassword';
import VerifyOTP from './Frontend/Components/VerifyOTP';
import Home from './Frontend/Components/Home Screen';
import Profile from './Frontend/Components/Profile';
import Icon from 'react-native-vector-icons'
const Stack = createNativeStackNavigator();
const Drawer = createDrawerNavigator();

const CustomDrawer = props => {

return (
<View style={{ flex: 1 }}>
<DrawerContentScrollView {...props}>
<View
style={{
alignSelf: "center",
padding: 20,

      }}
    >
      <View>
        <Text style={{ fontSize: 18, fontWeight: "bold" }}> App</Text>
      </View>

    </View>
    <DrawerItemList {...props} />
    <LogoutButton iconName="sign-out" color="black" />
  </DrawerContentScrollView>

</View>
);
};

function DrawerRoutes() {

return (

<Drawer.Navigator initialRouteName="Home"

  drawerContent={props => <CustomDrawer {...props} />}
>

  <Drawer.Screen name="Home" component={Home}

    options={{
      headerShown: true,
      
      title: 'Home',
      drawerIcon: ({ focused, size }) => (
        <Icon
          name="home"
          size={size}
          color='black'

        />

      )
    }}
  />
  <Drawer.Screen name="Profile" component={Profile}

    options={{
      headerShown: true,
      title: 'Profile',
      drawerIcon: ({ focused, size }) => (
        <Icon
          name="user"
          size={size}
          color='black'

        />

      )
    }}
  />




</Drawer.Navigator>
);
}

export default function App() {
return (
<ApplicationProvider {...eva} theme={eva.light}>

<NavigationContainer>
<Stack.Navigator>
          <Stack.Screen name="Registration" component={Reg_form} />
          <Stack.Screen name="Log In" component={Login_form}/>
          <Stack.Screen name="Forgot" component={ForgotPassword}/>
          <Stack.Screen name="Password" component={ChangePassword}/>
          <Stack.Screen name="Verification" component={VerifyOTP}/>
          <Stack.Screen name="Home" component={Home}  />
          <Stack.Screen name="DrawerRoutes" component={DrawerRoutes}/>
</Stack.Navigator>

); }
When we log in to the App, then navigate into the Home Screen. Then we have not visible Drawer navigation
